ELSpace/Screens/Hub/Activity/Type/Chooser/ChooserActivityTypesViewModel.swift

```diff
@@ -0,0 +1,40 @@
+import UIKit
+import RxSwift
+
+class ChooserActivityTypesViewModel: ChooserActivityTypesViewModeling {
+
+    init() {
+        configureViewModels()
+    }
+
+    lazy var activityTypeViewModels: [ActivityTypeViewModeling] = [timeReportViewModel,
+                                                                   vacationViewModel,
+                                                                   dayOffViewModel,
+                                                                   sickLeaveViewModel,
+                                                                   conferenceViewModel]
+
+    // MARK: - Private
+
+    private let timeReportViewModel = ActivityTypeViewModel(type: .timeReport)
+    private let vacationViewModel = ActivityTypeViewModel(type: .vacation)
+    private let dayOffViewModel = ActivityTypeViewModel(type: .dayOff)
+    private let sickLeaveViewModel = ActivityTypeViewModel(type: .sickLeave)
+    private let conferenceViewModel = ActivityTypeViewModel(type: .conference)
+
+    private let disposeBag = DisposeBag()
+    private weak var lastSelectedViewModel: ActivityTypeViewModeling?
```

Comment:
Why it's weak? You already handle references to all ActivityTypeViewModel in this file

Reply:
Weak is to avoid retain cycle and the reference is necessary for lastly-selected-type unselection

Reply:
But there is no retain cycle here

Reply:
Will explain it tomorrow